plugin/core/sessions.py

```diff
@@ -469,23 +470,62 @@ def can_start(cls, window: sublime.Window, initiating_view: sublime.View,
         """
         return None
 
+    @classmethod
+    def on_pre_start(cls, window: sublime.Window, initiating_view: sublime.View,
+                     workspace_folders: List[WorkspaceFolder], resolved: ResolvedStartupConfig) -> Optional[str]:
+        """
+        Callback invoked just before the language server subprocess is started. This is the place to do last-minute
+        adjustments to your "command" or "initializationOptions" in the passed-in "resolved" argument. You can also
+        choose to return a custom working directory, but consider that a language server shouldn't care about the
+        working directory.
+
+        :param      window:             The window
+        :param      initiating_view:    The initiating view
+        :param      workspace_folders:  The workspace folders
+        :param      resolved:           The resolved configuration
+
+        :returns:   A desired working directory, or None if you don't care
+        """
+        return None
+
+    @classmethod
+    def on_post_start(cls, window: sublime.Window, initiating_view: sublime.View,
+                      workspace_folders: List[WorkspaceFolder], resolved: ResolvedStartupConfig) -> None:
+        """
+        Callback invoked when the subprocess was just started.
+
+        :param      window:             The window
+        :param      initiating_view:    The initiating view
+        :param      workspace_folders:  The workspace folders
+        :param      resolved:           The resolved configuration
+        """
+        return None
+
     def __init__(self, weaksession: 'weakref.ref[Session]') -> None:
         """
-        Constructs a new instance.
+        Constructs a new instance. Your instance is constructed after a response to the initialize request.
 
         :param      weaksession:  A weak reference to the Session. You can grab a strong reference through
                                   self.weaksession(), but don't hold on to that reference.
         """
         self.weaksession = weaksession
 
+    def on_pre_window_settings(self, settings: DottedDict) -> None:
+        """
+        Override to alter the server settings for the workspace/didChangeConfiguration notification.
+
+        :param      settings:      The settings about to be sent to the language server
+        """
+        return None
+
     def on_workspace_configuration(self, params: Dict, configuration: Any) -> None:
         """
         Override to augment configuration returned for the workspace/configuration request.
 
         :param      params:         A ConfigurationItem for which configuration is requested.
         :param      configuration:  The resolved configuration for given params.
         """
-        pass
+        return None
```

Comment:
It's not really expected to receive anything from this API so I think `pass` makes more sense semantically.